### DISCUSSION TITLE
platform-espressif32 library to use

### DISCUSSION THREAD

**Discussion by proddy** [emsesp/EMS-ESP32#1330] [General]
@MichaelDvP , Jason has updated his [library](https://github.com/tasmota/platform-espressif32/releases/tag/2023.10.04) to 2023.10.04 and removed the WifiClientSecure class which we need for SSL in espMqttClient. So the compile breaks. Shall we fork or hardcode to an earlier release?

**Comment by MichaelDvP** [comment]
Also the espressif arduino with platform 6.4.0 gives now error, i think it's something like in 2.0.11 the `WiFi.h`, because the header is still in the [code](https://github.com/espressif/arduino-esp32/tree/master/libraries/WiFiClientSecure/src). 
For the esp32 without mqtt-TSL i've just modified the mqtt library. See https://github.com/MichaelDvP/EMS-ESP32/commit/b4e266f128074aba151469f27584263a7578c313 but for S3 it fails.

**Reply by proddy** [reply]
ok, that would work. Do you think some of the connectivity issues we've been seeing are related to changes in the arduino core wifi library? We never had those issues with Mesh for example back on 2.0.8

**Reply by MichaelDvP** [reply]
It does not work for S3 with SSL-mqtt, in actual arduino WiFiClientSecure is not fond, in older WiFi.h is not found in ETH.h. I have disabled the mqtt-SSL option and use tasmota platform in my dev2.
> We never had those issues with Mesh for example back on 2.0.8

We can make a testbuild, maybe with platform 5.2.0, that may fit in the 4M. But i expect no change.

**Reply by proddy** [reply]
ok, if you could do a PR with your latest dev and then I'll continue to build off that

**Reply by MichaelDvP** [reply]
There are some changes for arduino 2.0.14/3.0.0 that will give error in 2.0.09 and older (like platform 5.2): commits:
https://github.com/emsesp/EMS-ESP32/commit/62eeca944bd0694c74c877263d14212ec0db9eb8 and https://github.com/emsesp/EMS-ESP32/commit/a3300e94a7c08b8b7a991e4ba74780d72f4d90ce
